include/swift/SIL/SILValue.h

```diff
@@ -343,6 +382,148 @@ class SILValue {
                        DeadEndBlocks *DEBlocks = nullptr) const;
 };
 
+/// A map from a ValueOwnershipKind that an operand can accept to a
+/// UseLifetimeConstraint that describes the effect that the operand's use has
+/// on the underlying value. If a ValueOwnershipKind is not in this map then
+/// matching an operand with the value results in an ill formed program.
```

Comment:
Andy and I spoke about this. I agree with him that this is the direction in which things need to move. I want to make those (related) change in a different PR since I am trying to keep this PR as close to a refactor as possible.

Reply:
Cool. Thanks @gottesmm .